### ISSUE TITLE
ImageSequence broken on non-stack SPIDER image

### ISSUE THREAD

**Issue by jmichalon** [python-pillow/Pillow#1740]
Using ImageSequence.Iterator on a non-stack SPIDER image leads to infinite loop. Looking into the code, EOFError is never raised because when the image isn't a stack, seek() returns gently without error.
This fixes the problem:
```
     def seek(self, frame):
         if self.istack == 0:
+            raise EOFError("attempt to seek in a non-stack file")
-            return
         if frame >= self._nimages:
             raise EOFError("attempt to seek past end of file")
```

**Comment by radarhere** [comment]
Hey. Thanks for reporting this problem.

If you'd like, you can create the PR directly, so that your code can be reviewed by someone for acceptance into the project. If you have any problems with this, let us know, and maybe we can help.

The other thing that would be great is a test to repeat the error, so that this doesn't get re-introduced in the future.

**Comment by jmichalon** [comment]
Here you are: #1741 :)
I've added a comment to the commit about a test case but with that memory thing I don't know how to handle…

**Comment by wiredfool** [comment]
It looks like it's effectively a DOS, so we should probably just have the tests for it named `Tests/check_spider_imagesequence.py`

**Comment by wiredfool** [comment]
Merged in #1794
